### DISCUSSION TITLE
Call Fire on a Scene?

### DISCUSSION THREAD

**Discussion by keeper2597** [leolabs/ableton-js#69] [Q&A] [answered]
Thanks for the excellent library.  My TS is weak, but I'm happy to contribute on stuff I can wrap my head around.  I'm trying to implement a way to fire a Scene.  I tried to replicate the code found in Clip to access a command but I must be missing something as it errors out every time.

First, am I approaching this the right way?

`export declare class Scene extends Namespace<GettableProperties, TransformedProperties, SettableProperties, ObservableProperties> {
    raw: RawScene;
    constructor(ableton: Ableton, raw: RawScene);

    fire(): Promise<void>;
}`

`Scene.prototype.fire = function () {
        return this.sendCommand("fire")
    };`

Or is there a simpler way to let commands pass through?  Then, is this how I would call the command?

`scenes[next_index].fire();`

Thanks!

**Comment by keeper2597** [comment, marked as answer]
Update - user error here.  Maybe I can clear something up for someone else finding this.  The way that works to call fire on a scene is this:

`scene[index].sendCommand("fire");`

Thanks again for this project.

**Reply by leolabs** [reply]
I've just published a new version that includes the `fire` command directly as a function on scenes. This way, you can just call `scene[index].fire()`.

**Reply by keeper2597** [reply]
Ah, even better.  Thanks!
